app/src/lib/stores/pull-request-store.ts

```diff
@@ -1,237 +1,301 @@
+    })
   }
 
   /** Loads all pull requests against the given repository. */
-  public async fetchAndCachePullRequests(
-    repository: Repository,
-    account: Account
-  ): Promise<void> {
-    const githubRepo = forceUnwrap(
-      'Can only refresh pull requests for GitHub repositories',
-      repository.gitHubRepository
-    )
-    const apiClient = API.fromAccount(account)
+  public refreshPullRequests(repo: GitHubRepository, account: Account) {
+    const dbId = forceUnwrap("Can't refresh PRs, no dbId", repo.dbID)
 
-    this.updateActiveFetchCount(githubRepo, Increment)
+    const currentOp = this.currentRefreshOperations.get(dbId)
 
-    try {
-      const apiResult = await apiClient.fetchPullRequests(
-        githubRepo.owner.login,
-        githubRepo.name,
-        'open'
-      )
+    if (currentOp !== undefined) {
+      return currentOp
+    }
 
-      await this.cachePullRequests(apiResult, githubRepo)
+    this.lastRefreshForRepository.set(dbId, Date.now())
+    this.emitIsLoadingPullRequests(repo, true)
 
-      const prs = await this.fetchPullRequestsFromCache(githubRepo)
+    const promise = this.fetchAndStorePullRequests(repo, account)
+      .catch(err => {
+        log.error(`Error refreshing pull requests for '${repo.fullName}'`, err)
+      })
+      .then(() => {
+        this.currentRefreshOperations.delete(dbId)
+        this.emitIsLoadingPullRequests(repo, false)
+      })
 
-      await this.pruneForkedRemotes(repository, prs)
+    this.currentRefreshOperations.set(dbId, promise)
+    return promise
+  }
 
-      this.emitUpdate(githubRepo)
-    } catch (error) {
-      log.warn(`Error refreshing pull requests for '${repository.name}'`, error)
-    } finally {
-      this.updateActiveFetchCount(githubRepo, Decrement)
+  /**
+   * Fetches pull requests from the API (either all open PRs if it's the
+   * first time fetching for this repository or all updated PRs if not).
+   *
+   * Returns a value indicating whether it's safe to avoid
+   * emitting an event that the store has been updated. In other words, when
+   * this method returns false it's safe to say that nothing has been changed
+   * in the pull requests table.
+   */
+  private async fetchAndStorePullRequests(
+    repo: GitHubRepository,
+    account: Account
+  ) {
+    const api = API.fromAccount(account)
+    const lastUpdatedAt = await this.db.getLastUpdated(repo)
+
+    // If we don't have a lastUpdatedAt that mean we haven't fetched any PRs
+    // for the repository yet which in turn means we only have to fetch the
+    // currently open PRs. If we have fetched before we get all PRs
+    // If we have a lastUpdatedAt that mean we have fetched PRs
+    // for the repository before. If we have fetched before we get all PRs
+    // that have been modified since the last time we fetched so that we
+    // can prune closed issues from our database. Note that since
+    // fetchPullRequestsUpdatedSince returns all issues modified _at_ or
```

Comment:
I can't find the function associated with this comment. Has this been renamed?

Reply:
Yup, I'll update